src/stk/ea/mutation/mutators/molecule/random_smarts.py

```diff
@@ -0,0 +1,121 @@
+"""
+Random Smarts
+=============
+
+"""
+
+import numpy as np
+
+
+from .mutator import MoleculeMutator
+from ...records import MutationRecord
+from ....molecule_records import MoleculeRecord
+from rdkit import Chem
+from .....molecular import BuildingBlock
+
+
+class RandomSmarts(MoleculeMutator):
+    """
+    Substitutes functional groups within building blocks.
+
+    This mutator takes a :class:`ConstructedMolecule` and substitutes
+    the existing building blocks with ones containing
+    new functionalities, as specified in the SMARTS strings.
+    Atoms in functional groups cannot be adjusted as these are used to
+    construct the :class: `ConstructedMolecule`, and will
+    result in an error being raised.
+    Only external functionalities can be replaced.
+    """
+    def __init__(
+        self,
+        query_smarts,
+        replacement_smarts,
+        is_replaceable,
+        replacement_fgs,
+        replacement_specifier='all',
+        name='RandomSmarts',
+        random_seed=None,
+    ):
+        """
+        Initialize a :class:`.RandomSmarts` instance.
+
+        Parameters
+        ----------
+        query_smarts : :class`string`
+            SMARTS string to match on the :class: `BuildingBlock`
+
+        replacement_smarts : :class`string`
+            SMARTS string to replace those in `query_smarts` on the
+            :class: `BuildingBlock`
+
+        is_replaceable : :class:`callable`
+            A function which takes a :class:`.BuildingBlock` and
+            returns ``True`` or ``False``. This function is applied to
+            every building block in the molecule being mutated.
+            Building blocks which returned ``True`` are liable for
+            substitution by one of the molecules in `building_blocks`.
+
+        replacement_specifier : :class:`str`, optional
+            Specifies the number of SMARTS replacements to make.
+            If `all`, all matching SMARTS will be replacemed.
+            If `one`, a single random SMARTS replacement will occur.
```

Comment:
Is there ever a case where we would sub only one actually? It seems dangerous to me because the user has no control over which bit of the building block gets substituted -> making it potentailly awkward to use.

Reply:
I've removed this argument following this, I agree with your point. Thanks for the suggestion about Enum, I'd never heard of that before!